src/main/java/it/gov/pagopa/merchant/repository/MerchantRepository.java

```diff
@@ -27,4 +27,6 @@ public interface MerchantRepository extends MongoRepository<Merchant, String>, M
 
     )
     Optional<Merchant> retrieveByInitiativeIdAndMerchantId(String initiativeId, String organizationId, String merchantId);
+
+    Optional<Merchant> findByMerchantId(String merchantId);
```

Comment:
MerchantId is supposed to be the @Id of the collection (it was a mistake the fact that it has not the annotation, Giuseppe has fixed it on its PR)

Wait for Giuseppe's PR, then remove this method and use findById method